src/Query.tsx

```diff
@@ -220,7 +221,10 @@ export default class Query<TData = any, TVariables = OperationVariables> extends
     }
 
     this.updateQuery(nextProps);
+
     if (nextProps.skip) return;
+
+    this.lastResult = this.queryObservable!.getLastResult();
```

Comment:
I'm curious if this is the best/only place we could capture the last result. In particular, since we're setting `this.lastResult = null` in `removeQuerySubscription`, and this line comes right before `this.startQuerySubscription()`, perhaps we should move the management of `this.lastResult` into `{start,remove}QuerySubscription`?

Reply:
I also see that `resubscribeToQuery` is calling `this.queryObservable!.getLastResult()` separately from this logic, so I wonder if `resubscribeToQuery` could be simplified now that we're tracking `this.lastResult`?

Reply:
The timing of capturing the last result is definitely tricky. I've tried it in a few different places, only to find out that it was either captured too soon or too late. Capturing it here addresses the timing issues, but moving it into `{start, remove}QuerySubscription` should work as well. I'll move it and we'll see - and I'll look into cleaning up `resubscribeToQuery` as well. Thanks!

Reply:
@benjamn I was able to move the storing of `lastResult` into `{start, remove}QuerySubscription`, but it took a bit of wrangling because `startQuerySubscription` is used when receiving new props and when the component first mounts. When it first mounts we want to ignore the `lastResult` no matter what (since this can cause problems if we mount a component, then unmount, then mount it again, and it had previous results that cause it to no longer render), so I tweaked the logic a bit to accommodate. This should be ready for another review - thanks!